intro-HPC/ch_MATLAB.tex

```diff
@@ -0,0 +1,165 @@
+%\shellcmd{module load MATLAB/2018a}
+\end{prompt}
+
+After loading the module, the \verb|mcc| command can be used. To get help on
+\verb|mcc|, you can run \verb|mcc -?|.
+
+To compile a standalone application, the \verb|-m|
+flag is used (the \verb|-v| flag means verbose ouput).
+To show how \verb|mcc| can be used, we use the \verb|magicsquare| example
+that comes with MATLAB.
+
+\begin{prompt}
+%\shellcmd{cp \$EBROOTMATLAB/extern/examples/compiler/magicsquare.m .}%
+%\shellcmd{mcc -mv magicsquare.m}%
+Opening log file:  %\homedir/java.log.34090%
+Compiler version: 6.6 (R2018a)
+Dependency analysis by REQUIREMENTS.
+Parsing file "%\homedir/magicsquare.m%"
+	(Referenced from: "Compiler Command Line").
+Deleting 0 temporary MEX authorization files.
+Generating file "%\homedir/readme.txt%".
+Generating file "run\_magicsquare.sh".
+\end{prompt}
+
+\subsection{Libraries}
+
+To compile a MATLAB program that \emph{needs a library}, you can use the
+\texttt{-I \emph{library\_path}} flag. This will tell the compiler to also
+look for files in \texttt{\emph{library\_path}}.
+
+It's also possible to use the \texttt{-a \emph{path}} flag. That will result in
+all files under the \texttt{\emph{path}} getting added to the final executable.
+
+\subsection{Memory issues during compilation}
+
+If you are seeing Java memory issues during the compilation of your MATLAB program
+on the login nodes, consider tweaking the default maximum heap size (128M) of Java
+using the \verb|_JAVA_OPTIONS| environment variable with:
+
+\begin{prompt}
+%\shellcmd{export \_JAVA\_OPTIONS="-Xmx64M"}%
+\end{prompt}
+
+The MATLAB compiler spawns multiple Java processes, and because of the default memory
+limits that are in effect on the login nodes, this might lead to a crash of the compiler
+if it's trying to create to many Java processes. If we lower the heap size, more
+Java processes will be able to fit in memory.
+
+Another possible issue is that the heap size is too small. This could result
+in errors like:
+
+\begin{prompt}
+Error: Out of memory
+\end{prompt}
+
+A possible solution to this is by setting the maximum heap size to be bigger:
+
+\begin{prompt}
+%\shellcmd{export \_JAVA\_OPTIONS="-Xmx512M"}%
+\end{prompt}
+
+\section{Multithreading}
+
+MATLAB can only use the cores in a single
+workernode (unless the Distributed Computing toolbox is used, see
+\url{https://nl.mathworks.com/products/distriben.html}).
+
+
+The amount of workers used by MATLAB for the parallel toolbox can be controlled
+via the \verb|parpool| function: \verb|parpool(16)| will use 16 workers.
+It's best to specify the amount of workers,
+because otherwise you might not harness the full compute power available (if you have
+too few workers), or you might negatively impact performance (if you have too much workers).
+By default, MATLAB uses a fixed number of workers (12).
+
+The best amount of workers is the amount of processors. This amount can be
+obtained via the following code:
```

Comment:
`You should use a number of workers that is equal to the number of cores you requested when submitting your job script (the `ppn` value, see <ref to section>). You can determine the right number of workers to use via the following code snippet in your MATLAB program:`